core/src/main/java/org/jdbi/v3/core/Handler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core;
+
+public interface Handler {
+    Handler STANDARD_HANDLER = new Handler() {
+        @Override
+        public <R, X extends Exception> R withHandle(Handle handle, HandleCallback<R, X> callback) throws X {
+            return callback.withHandle(handle);
+        }
+    };
+
+    /**
+     * Call a callback with a Handle
+     *
+     * @param handle the handle to the database
+     * @param callback a callback which will receive the open handle
+     *
+     * @return the value returned by the callback.
+     *
+     * @throws X any exception thrown by the callback.
+     * @see Jdbi#withHandle(HandleCallback)
+     */
+    <R, X extends Exception> R withHandle(Handle handle,
```

Comment:
Adding the ability to decorate `withHandle` seems fine.

One thought: as this is right now, the `Handler` will remain in the stack trace, even if it does nothing.

An alternate implementation would be `HandleCallback<R, X> decorate(HandleCallback<R, X> callback)`. Then, the default implementation would just `return callback;` and the stack frame would not show up in debuggers or profilers when it does nothing.

What do you think? I am fine with this approach too if you don't like the suggestion. cc @hgschmie

Reply:
Yeah, that's better. I've pushed that change.